ledger/src/shredder.rs

```diff
@@ -151,305 +136,19 @@ impl Shredder {
         Vec<Shred>, // data shreds
         Vec<Shred>, // coding shreds
     ) {
-        if merkle_variant {
-            return self
-                .make_merkle_shreds_from_entries(
-                    keypair,
-                    entries,
-                    is_last_in_slot,
-                    chained_merkle_root,
-                    next_shred_index,
-                    next_code_index,
-                    reed_solomon_cache,
-                    stats,
-                )
-                .partition(Shred::is_data);
-        }
-        let data_shreds =
-            self.entries_to_data_shreds(keypair, entries, is_last_in_slot, next_shred_index, stats);
-        let coding_shreds = Self::data_shreds_to_coding_shreds(
+        assert!(merkle_variant, "Legacy shreds are not supported");
```

Comment:
This assert means the following PR must land first: https://github.com/anza-xyz/agave/pull/7107

Reply:
Yep, 100%. Splitting up PRs correctly is tricky.